homeassistant/components/otbr/config_flow.py

```diff
@@ -101,27 +108,66 @@ async def _connect_and_set_dataset(self, otbr_url: str) -> None:
                 )
             await api.set_enabled(True)
 
+    async def _is_extended_address_configured(self, extended_address: bytes) -> bool:
+        """Return True if another config entry's OTBR has the same extended address."""
+        if DATA_OTBR not in self.hass.data:
+            _LOGGER.debug("No OTBR")
+            return False
+        for data in self.hass.data[DATA_OTBR].values():
+            try:
+                other_extended_address = await data.get_extended_address()
+            except HomeAssistantError:
+                _LOGGER.debug(
+                    "Could not read extended address from %s", data.url, exc_info=True
+                )
+                continue
+            _LOGGER.debug(
+                "extended address for url %s: %s",
+                data.url,
+                other_extended_address.hex(),
+            )
+            if extended_address == other_extended_address:
+                return True
+        return False
+
+    async def _connect_and_configure_router(self, otbr_url: str) -> bytes:
+        """Connect to the router and configure it if needed.
+
+        Will raise if the router's extended address is in use by another confi entry.
+        Returns the router's extended address.
+        """
+        api = python_otbr_api.OTBR(otbr_url, async_get_clientsession(self.hass), 10)
+        extended_address = await self._get_extended_address(api)
+        _LOGGER.debug("extended_address: %s", extended_address.hex())
+
+        if await self._is_extended_address_configured(extended_address):
+            raise AlreadyConfigured
+
+        await self._set_dataset(api, otbr_url)
+
+        return extended_address
+
     async def async_step_user(
         self, user_input: dict[str, str] | None = None
     ) -> ConfigFlowResult:
         """Set up by user."""
-        if self._async_current_entries():
-            return self.async_abort(reason="single_instance_allowed")
-
         errors = {}
 
         if user_input is not None:
             url = user_input[CONF_URL].rstrip("/")
             try:
-                await self._connect_and_set_dataset(url)
+                extended_address = await self._connect_and_configure_router(url)
+            except AlreadyConfigured:
+                errors["base"] = "already_configured"
```

Comment:
I guess we need to add a string/translation somewhere for `already_configured`? :thinking: